### DISCUSSION TITLE
Get biggest documents

### DISCUSSION THREAD

**Discussion by RvZEU** [ravendb/ravendb#17670] [Q&A] [answered]
Hi,

does anybody know how to sort by document size? I would like to know which documents are bigger then X.

Also when you delete all the contents of a collection, how long does it take before that free space is allocated to the HD?

Thank you in advance.

**Comment by ppekrol** [comment, marked as answer]
Hi,

You can create a static index and just query it:

Map:
```
from doc in docs
select new {
    SizeInBytes = doc.BlittableJson.Size
}
```

When it comes to free space, we are not releasing it at any time, it is being reused for future needs. If you need to deallocate then you would have to perform compaction on the database/indexes.
